how-to-guides/hpo/notebooks/Neptune_HPO.ipynb

```diff
@@ -0,0 +1,647 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "learning_rates = [0.05, 0.1, 0.5]  # learning rate choices"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class BaseModel(nn.Module):\n",
+    "    def __init__(self, input_size, num_classes):\n",
+    "        super(BaseModel, self).__init__()\n",
+    "        self.fc1 = nn.Linear(input_size, 128)\n",
+    "        self.fc2 = nn.Linear(128, 64)\n",
+    "        self.fc3 = nn.Linear(64, num_classes)\n",
+    "        self.relu = nn.ReLU()\n",
+    "        self.dropout = nn.Dropout(0.5)\n",
+    "\n",
+    "    def forward(self, x):\n",
+    "        x = self.relu(self.fc1(x))\n",
+    "        x = self.dropout(x)\n",
+    "        x = self.relu(self.fc2(x))\n",
+    "        x = self.dropout(x)\n",
+    "        x = self.fc3(x)\n",
+    "        return x\n",
+    "\n",
+    "\n",
+    "criterion = nn.CrossEntropyLoss()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_tfms = {\n",
+    "    \"train\": transforms.Compose(\n",
+    "        [\n",
+    "            transforms.ToTensor(),\n",
+    "        ]\n",
+    "    )\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trainset = datasets.MNIST(\n",
+    "    root=\"mnist\",\n",
+    "    train=True,\n",
+    "    download=True,\n",
+    "    transform=data_tfms[\"train\"],\n",
+    ")\n",
+    "\n",
+    "trainloader = torch.utils.data.DataLoader(\n",
+    "    trainset,\n",
+    "    batch_size=parameters[\"batch_size\"],\n",
+    "    shuffle=True,\n",
+    "    num_workers=0,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Log metadata across HPO trials into a single run"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Initialize Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = BaseModel(\n",
+    "    input_size,\n",
+    "    parameters[\"n_classes\"],\n",
+    ").to(parameters[\"device\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a global Neptune run"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from random import random\n",
+    "\n",
+    "run = Run(\n",
+    "    family=\"hpo\",\n",
+    "    run_id=f\"hpo-{random()}\",\n",
+    ")\n",
```

Comment:
Experiment name?

Reply:
Experiments not needed for this use-case

Reply:
Hmm, alright. I was just under the impression that we should treat experiments as the main entity and runs as little more than "experiment remnants". But let's maybe revisit this once we have the new data model.